sql/compression_defaults.sql

```diff
@@ -0,0 +1,291 @@
+        INNER JOIN LATERAL
+            (select * from unnest(i.indkey) with ordinality) a(attnum, pos) ON (TRUE)
+        WHERE a.pos <= i.indnkeyatts
+        GROUP BY 1
+    )
+    SELECT
+      a.attname INTO _segmentby
+    FROM
+      index_attr i
+    INNER JOIN
+      pg_attribute a on (a.attnum = i.attnum AND a.attrelid = relation)
+    LEFT JOIN
+      pg_catalog.pg_attrdef ad ON (ad.adrelid = relation AND ad.adnum = a.attnum)
+    LEFT JOIN
+      pg_statistic s ON (s.staattnum = a.attnum and s.starelid = relation)
+    WHERE
+      a.attname NOT IN (SELECT column_name FROM _timescaledb_catalog.dimension d WHERE d.hypertable_id = _hypertable_row.id)
+      AND s.stadistinct is null
+      AND a.attidentity = '' AND (ad.adbin IS NULL OR pg_get_expr(adbin, adrelid) not like 'nextval%')
+    ORDER BY i.pos
+    LIMIT 1;
+
+    IF FOUND THEN
+            return json_build_object(
+            'columns', json_build_array(_segmentby),
+            'confidence', 5,
+            'message',  'Please make sure '|| _segmentby||' is not a unique column and appropriate for a segment by');
+    END IF;
+
+
+    --are there any indexed columns that are not dimemsions and are not serial/identity?
+    with index_attr as (
+        SELECT
+        a.attnum, min(a.pos) as pos
+        FROM
+            (select indkey, indnkeyatts from pg_catalog.pg_index where indisunique and indrelid = relation) i
+        INNER JOIN LATERAL
+            (select * from unnest(i.indkey) with ordinality) a(attnum, pos) ON (TRUE)
+        WHERE a.pos <= i.indnkeyatts
+        GROUP BY 1
+    )
+    SELECT
+      count(*) INTO STRICT _cnt
+    FROM
+      index_attr i
+    INNER JOIN
+      pg_attribute a on (a.attnum = i.attnum AND a.attrelid = relation)
+    LEFT JOIN
+      pg_catalog.pg_attrdef ad ON (ad.adrelid = relation AND ad.adnum = a.attnum)
+    WHERE
+      a.attname NOT IN (SELECT column_name FROM _timescaledb_catalog.dimension d WHERE d.hypertable_id = _hypertable_row.id)
+      AND a.attidentity = '' AND (ad.adbin IS NULL OR pg_get_expr(adbin, adrelid) not like 'nextval%');
+
+    IF _cnt > 0 THEN
+        return json_build_object(
+            'columns', json_build_array(),
+            'confidence', 0,
+            'message',  'Several columns are potential segment by candidates. Please use the segment_by option to explicitly specify the segment_by column');
+    ELSE
+        return json_build_object(
+            'columns', json_build_array(),
+            'confidence', 5,
+            'message',  'You do not have any indexes on columns that can be used for segment_by and thus we are not using segment_by for compression. Please make sure you are not missing any indexes');
+    END IF;
+END
+$BODY$ SET search_path TO pg_catalog, pg_temp;
+
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.get_orderby_defaults(
+    relation regclass, segment_by_cols text[]
```

Comment:
Maybe you want to have a default for the `segment_by_cols`?

Reply:
This is not meant as a user-facing function, so I don't think we need defaults